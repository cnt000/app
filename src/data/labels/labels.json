{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright © 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B\nIscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse\nNegli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9\nDicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001\nVAB",
  "mailToLink": "mailto:info@pungilandia.com",
  "privacyLink": "http://www.pungilandia.com",
  "addToCart": "Aggiungi al carrello",
  "searchPlants": "Cerca piante",
  "searchPlantsImage": "http://www.pungilandia.com/php/images/lo/cotyl_mediop.jpg",
  "hintsFromPungilandiaTitle": "I Consigli di Pungilandia",
  "articleBy": "A cura del dott. Mimmo Rocchi",
  "articleContent": "consigliColtivazione.md",
  "mimmoImage": "http://www.pungilandia.com/media/MIMMO.JPG",
  "menuChiSiamo": "Chi siamo",
  "menuListaPiante": "Lista piante",
  "menuNovita": "Novità",
  "menuProdottiSpeciali": "Prodotti speciali",
  "menuConsigliPerColtivare": "Consigli per coltivare",
  "menuOrdini": "Ordini",
  "menuFotoShow": "Foto show",
  "menuEmail": "E-mail"
}
