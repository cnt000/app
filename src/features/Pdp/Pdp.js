import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import LabelsContext from '../Labels';
import styles from './Pdp.module.css';

const Pdp = ({ name, image, size, price }) => {
  const { addToCart } = useContext(LabelsContext);
  const imageFullUrl = `http://www.pungilandia.com/php/${image}`;
  return (
    <main className={styles.pdpContent} role="main">
      <div className={styles.header}></div>
      <div className={styles.mainImage}>
        <img src={imageFullUrl} alt={name} />
      </div>
      <div className={styles.product}>
        <h2 className={styles.name}>{name}</h2>
        <div className={styles.dimension}>{size}</div>
        <div className={styles.price}>{price} â‚¬</div>
        <button className={styles.addToCart}>{addToCart}</button>
      </div>
    </main>
  );
};

Pdp.propTypes = {
  name: PropTypes.string,
  image: PropTypes.string,
  size: PropTypes.string,
  price: PropTypes.string,
  addToCart: PropTypes.string,
};

export default Pdp;
