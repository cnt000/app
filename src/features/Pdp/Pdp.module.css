.pdpContent {
  grid-area: content;
  padding: 0.4rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}

.product {
  width: 100%;
  margin-bottom: 1.6rem;
}

.name,
.dimension,
.price,
.addToCart {
  margin-top: 0;
  margin-bottom: 0.8rem;
}

.name {
  font-size: 22px;
  margin-bottom: 1.6rem;
}

.infoBox {
  font-size: 16px;
}

.mainImage {
  margin: 1rem auto;
  text-align: center;
}

.mainImage img {
  width: 100%;
  max-width: 500px;
}

.productInfo {
  text-align: center;
  margin: 1rem 0;
}

.addToCart {
  margin: var(--base-gap) auto calc(var(--base-gap) * 2) auto;
  background-color: var(--medium-green);
  color: white;
  padding: calc(var(--base-gap) * 4);
  border-radius: 3px;
  border: 1px solid white;
  cursor: pointer;
  width: 100%;
}

.infoBox {
  display: flex;
  justify-content: space-between;
}

@media only screen and (min-width: 768px) {
 .pdpContent {
   width: 460px;
  }
  .infoBox {
    margin-bottom: 1.4rem;
  }
}

@media only screen and (min-width: 1024px) {
 .pdpContent {
    display: grid;
    grid-template-columns: 10% 60% 30%;
    width: 100%;
  }
  .mainImage {
    margin-left: 0;
    margin-right: 2rem;
  }
  .mainImage img {
    max-width: 580px;
  }
  .product {
    width: 80%;
  }
  .name {
    font-size: 2rem;
  }
  .infoBox {
    font-size: 1.2rem;
  }
  .addToCart {
    font-size: 1rem;
  }
  .infoBox {
    flex-direction: column;
  }
}
