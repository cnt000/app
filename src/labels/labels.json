{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright Â© 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B<br>Iscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse<br>Negli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9<br>Dicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001<br>VAB"
}
