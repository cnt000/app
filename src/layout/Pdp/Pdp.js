import React from 'react';
import styles from './Pdp.module.css';

const Pdp = ({ productName, productImage, vaseDiameter, price, addToCart }) => (
  <main className={styles.pdpContent} role="main">
    <div className={styles.header}></div>
    <div className={styles.mainImage}>
      <img src={productImage} alt={productName} />
    </div>
    <div className={styles.product}>
      <h2 className={styles.name}>{productName}</h2>
      <div className={styles.dimension}>{vaseDiameter}</div>
      <div className={styles.price}>{price}</div>
      <button className={styles.addToCart}>{addToCart}</button>
    </div>
  </main>
);

export default Pdp;
