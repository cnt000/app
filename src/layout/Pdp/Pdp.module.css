.pdpContent {
  grid-area: content;
  padding: 0.4rem;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.name,
.dimension,
.price,
.addToCart {
  margin-top: 0;
  margin-bottom: 0.8rem;
}

.name {
  font-size: 22px;
}

.price {
  font-size: 16px;
}

.mainImage {
  margin: 1rem auto;
  text-align: center;
}

.mainImage img {
  width: 100%;
  max-width: 500px;
}

.productInfo {
  text-align: center;
  margin: 1rem 0;
}

.addToCart {
  margin: 0 auto;
  margin: 1rem auto 2rem auto;
  width: 50%;
  height: 2rem;
}

@media only screen and (min-width: 1025px) {
 .pdpContent {
    display: grid;
    grid-template-columns: 10px 1fr 30%;
  }
}
