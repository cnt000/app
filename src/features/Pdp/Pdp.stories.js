import React from 'react';

import Pdp from './Pdp';
import labels from '../../data/labels/labels.json';
import LabelsContext from '../../features/Labels';
import product from '../../data/products/clavifolius.json';

export default {
  component: Pdp,
  title: 'Pdp',
};

export const Default = () => (
  <LabelsContext.Provider value={labels}>
    <Pdp {...product} />
  </LabelsContext.Provider>
);
