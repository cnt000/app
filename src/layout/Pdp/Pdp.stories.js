import React from 'react';

import Pdp from './Pdp';
import labels from '../../data/labels/labels.json';
import product from '../../data/products/clavifolius.json'

export default {
  component: Pdp,
  title: 'Pdp',
};

export const Default = () => <Pdp {...labels} {...product} />;
