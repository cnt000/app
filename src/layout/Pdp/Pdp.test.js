import React from 'react';
import renderer from 'react-test-renderer';

import LabelsContext from '../../features/labels';
import labels from '../../data/labels/labels.json';
import Pdp from './Pdp';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <LabelsContext.Provider value={labels}>
        <Pdp />
      </LabelsContext.Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
