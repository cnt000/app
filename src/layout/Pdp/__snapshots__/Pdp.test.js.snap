// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="pdpContent"
  role="main"
>
  <div
    className="header"
  />
  <div
    className="mainImage"
  >
    <img />
  </div>
  <div
    className="product"
  >
    <h2
      className="name"
    />
    <div
      className="dimension"
    />
    <div
      className="price"
    />
    <button
      className="addToCart"
    />
  </div>
</main>
`;
